src/main/scala/com/fulcrumgenomics/bam/FlagStat.scala

```diff
@@ -0,0 +1,161 @@
+  * pass-filter or reads that do not pass filter, but not both.
+  *
+  * See the [SAMTools flagstat documentation](http://www.htslib.org/doc/samtools.html)
+  *
+  * @param pf true if the metric is for pass-filter reads, false for reads that did not pass-filter.
+  * @param reads the total number of reads/records.
+  * @param mapped the number of mapped reads/records.
+  * @param paired the number of reads/records that were paired.
+  * @param mapped_in_pair the number of mapped paired reads/records with its mate mapped.
+  * @param proper_paired the number of paired reads/records that were properly paired.
+  * @param singleton the number of paired reads/records that were mapped but its mate was not mapped.
+  * @param read_one the number of first of pair reads/records.
+  * @param read_two the number of second of pair reads/records.
+  * @param duplicate the number of duplicate reads/records.
+  * @param inter_chrom the number of mapped paired reads/records whose mate maps to a different chromosome.
+  * @param inter_chrom_mapq5 the number of mapped paired reads/records with mapping quality at least five whose mate maps to a different chromosome.
+  * @param secondary the number of secondary reads/records.
+  * @param supplementary the number of supplementary reads/records.
+  */
+case class FlagStatMetric
+(pf: Boolean,
+ var reads: Long = 0,
+ var mapped: Long = 0,
+ var paired: Long = 0,
+ var mapped_in_pair: Long = 0,
+ var proper_paired: Long = 0,
+ var singleton: Long = 0,
+ var read_one: Long = 0,
+ var read_two: Long = 0,
+ var duplicate: Long = 0,
+ var inter_chrom: Long = 0,
+ var inter_chrom_mapq5: Long = 0,
+ var secondary: Long = 0,
+ var supplementary: Long = 0
+) extends Metric
+
+@clp(group = ClpGroups.SamOrBam, description=
+  """|Collects metrics from the FLAG field in a SAM or BAM file.
+  """)
+class FlagStat
+( @arg(flag='i', doc="Input SAM or BAM file.") val input: PathToBam,
+  @arg(flag='o', doc="The output file in which metrics will be stored.") val output: Option[PathToBam] = None
+) extends FgBioTool with LazyLogging {
+
+  override def execute(): Unit = {
+    val metrics = Seq(true, false).map(pf => pf -> FlagStatMetric(pf=pf)).toMap
+    val in      = SamSource(input)
+
+    in.foreach { rec => FlagStat.updateMetric(metrics(rec.pf), rec) }
+    in.safelyClose
+
+    // Output to file, if given
+    output.foreach { out => Metric.write(out, metrics.values.toSeq.sortBy(_.pf).reverse) } // NB: write pf reads first
+
+    // Output the same format as SAMTools
+    System.out.print(FlagStat.formatInSamtools(metrics))
+  }
+}
+
+object FlagStat {
+  /** Updates the metric using the given record. */
+  private[bam] def updateMetric(metric: FlagStatMetric, rec: SamRecord): Unit = {
+    require(metric.pf == rec.pf, "The metric PF value did not match the record's pf value")
+    metric.reads += 1
+    if (rec.secondary) metric.secondary += 1
+    else if (rec.supplementary) metric.supplementary += 1
```

Comment:
Why is this and the next line an `else if`.  A record can be both secondary and supplementary.  And it can definitely be secondary or supplementary and also paired.